whipper/command/cd.py

```diff
@@ -475,6 +497,11 @@ def _ripIfNotRipped(number):
                 continue
             _ripIfNotRipped(i + 1)
 
+        if (self.options.fetch_cover_art == "embed" and
+                self.coverArtPath is not None):
```

Comment:
Seems like some kind of `with …` or `try: … finally: …` clause would be more appropriate, since otherwise this would potentially leave stray files lying around in case of crashes etc. That might be something for a future improvement though.

Reply:
Could you draft an example?
For the pull request just adding a comment with a `TODO`/`FIXME` tag to avoid forgetting this may be enough.

Reply:
Is there any update on this ?

Reply:
> Is there any update on this ?

I'm waiting for @Freso's reply.

Reply:
I think the idea would be:

```
try:
   download image somewhere
except:
  if os.exists(downloaded_file):
    os.remove(downloaded file)
```

To always clean up the cover art image.

Of course, you could also keep it in ram, no need to save it to disk per se. Then you could use StringIO for this:

```
        pic = Picture()
        pic.data = some_stringio_cover_art.read()
```

Or, if it's really just a set of bytes, you won't need the string io.

```
        pic = Picture()
        pic.data = cover_art_data
```

Reply:
I've checked Pillow's documentation and it seems one should pass a [`BytesIO`](https://docs.python.org/3.5/library/io.html#io.BytesIO) object to the `Image.open()` function. I think that's going to work for `mutagen` too.

Reply:
> I think the idea would be:
> ```python
> try:
>    download image somewhere
> except:
>   if os.exists(downloaded_file):
>     os.remove(downloaded file)
> ```

Isn't it necessary for that to work that the `try` block must cover almost every instruction of `_ripIfNotRipped` with a generic `except` catchall?

Reply:
Yes, it'd have to capture a bit part of the rip. As for mutagen, seeing that `.data` is assigned with `something.read()`, I assume that it just wants bytes.

Reply:
Added `NOTE` comment to code.